flink-java/src/main/java/org/apache/flink/api/java/io/CollectionInputFormat.java

```diff
@@ -130,8 +130,14 @@ public String toString() {
 			if (elem == null) {
 				throw new IllegalArgumentException("The collection must not contain null elements.");
 			}
-			
-			if (!viewedAs.isAssignableFrom(elem.getClass())) {
+
+			// The second part of the condition is a workaround for the situation that can arise from eg.
+			// "env.fromElements((),(),())"
+			// In this situation, UnitTypeInfo.getTypeClass returns void.class (when we are in the Java world), but
+			// the actual objects that we will be working with, will be BoxedUnits.
+			if (!viewedAs.isAssignableFrom(elem.getClass()) &&
```

Comment:
I agree that this is not elegant, but I don't see any better way.

UnitTypeInfo.getTypeClass can't return classOf[BoxedUnit], because TypeInformation<T>.getTypeClass() returns Class<T>, and here T is Unit.

I already had a test for this in TypeInformationGenTest.testUnit. Now I also added a comment, that explains that those two lines in the test are testing this condition.

By the way, if you really dislike this condition, then I can just remove it. The rest of the PR already solves my original problem, so I can live without "fromElements((),(),())" working.